### ISSUE TITLE
grey_kindOfClass doesn't accept a String

### ISSUE THREAD

**Issue by bootstraponline** [google/EarlGrey#104]
>   |  |--<_UIAlertControllerView:0x7a7b4600; AX=N; AX.label='Login Failure'; 

```swift
EarlGrey().selectElementWithMatcher(grey_allOfMatchers(
    grey_accessibilityLabel("Login Failure"),
    grey_kindOfClass("_UIAlertControllerView")))
```

Why can't I use a string for `grey_kindOfClass`? I can't reference _UIAlertControllerView directly as it appears to be a private class.

**Comment by bootstraponline** [comment]
`NSClassFromString("_UIAlertControllerView")` appears to do what I want but the match still fails. 

> Reason: Assertion 'assertWithMatcher:matcherForSufficientlyVisible(>=0.750000)' was not performed because no UI element matching (((respondsToSelector(isAccessibilityElement) && isAccessibilityElement) && accessibilityLabel("Login Failure")) && kindOfClass("_UIAlertControllerView"))

**Comment by khandpur** [comment]
Can you check if such an element exists in the UI hierarchy and whether it has accessibility label set to "Login Failure"

**Comment by bootstraponline** [comment]
> |  |--<_UIAlertControllerView:0x7a67cab0; AX=N; AX.label='Login Failure'; AX.frame={{25, 176}, {270, 128}}; AX.activationPoint={160, 240}; AX.traits='UIAccessibilityTraitNone'; AX.focused='N'; frame={{25, 176}, {270, 128}}; opaque; alpha=1>

>  <UILabel:0x7a6848e0; AX=Y; AX.label='Login Failure'; AX.frame={{41, 196}, {238, 20.5}}; AX.activationPoint={160, 206.25}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{16, 20}, {238, 20.5}}; opaque; alpha=1; interaction=N; text='Login Failure'

I see two entries. I can match the UILabel. I can't match on `_UIAlertControllerView`.

**Comment by bootstraponline** [comment]
If `AX=N` then EarlGrey shouldn't complain that there are multiple matches for `AX.label='Login Failure';`, right?

**Comment by bootstraponline** [comment]
```
). Please use selection matchers to narrow the selection down to single element." UserInfo={NSLocalizedDescription=Multiple elements were matched: (
    "<UILabel:0x7feeb080; AX=Y; AX.label='Login Failure'; 
      AX.frame={{41, 208}, {238, 20.5}}; AX.activationPoint={160, 218.25}; 
      AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; 
      frame={{16, 20}, {238, 20.5}}; opaque; alpha=1; interaction=N; text='Login Failure'>",
    "<UILabel:0x7be1f580; AX=Y; AX.label='Login Failure'; 
      AX.frame={{41, 196}, {238, 20.5}}; AX.activationPoint={160, 206.25}; 
      AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N';
      frame={{16, 20}, {238, 20.5}}; opaque; alpha=1; interaction=N; text='Login Failure'>"
). Please use selection matchers to narrow the selection down to single elemen
```

It looks like there's no way to simply say pick the first match. The elements are identical. I don't see multiple `Login Failures` in the screenshot. In the view dump, there's only one `UILabel`, the other is a `_UIAlertControllerView`.

**Comment by tirodkar** [comment]
Did you try adding `grey_interactable()` or `grey_sufficientlyVisible()` in your matcher?

**Comment by khandpur** [comment]
EG allows you to match on both accessible and non-accessible elements (if you use non accessibility matchers). I think here the problem is that _UIAlertControllerView isn't accessible but has an accessibility label set so it shouldn't be picked by EarlGrey because you're using grey_accessibilityLabel to match on something that's not accessible.

As for the Login Failures, it's picking up two different elements with different memory addresses. **Are these elements somehow duplicated in the app**?
1. UILabel: 0x7feeb080
2. UILabel: 0x7be1f580

Their frames are the same `frame={{16, 20}, {238, 20.5}}` but their accessibilityActivationPoints are different: `AX.activationPoint={160, 206.25}`

**Comment by bootstraponline** [comment]
The test passes if I add `grey_interactable()`. Thanks for the tip. I thought `interaction=N` based on the logs means it isn't intractable but somehow it matches.

> Are these elements somehow duplicated in the app?

Probably. This is looking like a bug in the app and not an EarlGrey issue. I'll talk with the devs tomorrow. Thanks for the help debugging.

I'm wondering if it's worth adding a FAQ entry about how to resolve identical duplicate element issues.

**Comment by khandpur** [comment]
Yup, that we should add it to FAQ :)
